File: spring-foundamental/spring-rest/src/main/java/com/omse/springrest/springrest/reository_rest_resourse/PersonController.java
package com.omse.springrest.springrest.reository_rest_resourse;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class PersonController {

    private final PersonService personService;

    public PersonController(PersonService personService) {
        this.personService = personService;
    }

    public ResponseEntity<?> findALl(){
        return ResponseEntity.ok().body(personService.findAll());
    }
}

File: kotlin/src/main/kotlin/com/omidmohebbise/kotlin/controller/PersonController.kt
package com.omidmohebbise.kotlin.controller

import com.omidmohebbise.kotlin.domain.Person
import com.omidmohebbise.kotlin.repository.PersonRepository
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class PersonController {

    val personRepository: PersonRepository

    constructor(personRepository: PersonRepository) {
        this.personRepository = personRepository
    }

    @GetMapping("/find-all")
    fun findAll(): ResponseEntity<List<Person>> {

        return ResponseEntity.ok().body(personRepository.findAll())
    }
    @GetMapping("/add")
    fun add(): ResponseEntity<Person> {

        return ResponseEntity.ok().body(personRepository.save(Person( "Omid Mohebbi")))
    }
}